cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -96,6 +106,76 @@
   }
 
   /**
+   * Migrate schedules in a given namespace from app metadata store to ProgramScheduleStoreDataset
+   * @param namespaceId the namespace with schedules to be migrated
+   * @param appMetaStore app metadata store with schedules to be migrated
+   * @return the lexicographically largest namespace id String with schedule migration completed
+   */
+  public String migrateFromAppMetadataStore(NamespaceId namespaceId, Store appMetaStore) {
```

Comment:
This method should read from `Store appMetaStore`, and write to `ProgramScheduleStoreDataset this`, right?